assets/javascripts/needleeditor.js

```diff
@@ -463,6 +500,17 @@ function saveNeedle(e) {
       addFlash('danger', '<strong>Unable to save needle:</strong><ul><li>' + errors.join('</li><li>') + '</li></ul>');
       return false;
   }
+
+  if ($("#property_workaround").prop("checked") && $("#input_workaround_desc").val() == "") {
+    var confirmMessage = "You set a workaround property for this needle, "
+                      + "but do not input description. "
```

Comment:
> but *no* input description

would be correct English. I also think it is more common to omit the `,` before `but`. I'm also not a fan of using string concatenation to emulate multi-line strings. I guess we're already using the ` \`-syntax in some places or just use one long string. (Not sure whether we already want to require ES6 template literals.)

Reply:
I think maybe it does not meet specifications when the string is too long and put them in one line, so I put them in multi-lines and connect with "+".  If you don't mind, I prefer one long string in line.

Reply:
There is not "tidy check" for JavaScript and hence not strict specifications. Just follow the overall style. Long lines are no problem.